### ISSUE TITLE
Store which sounds are downloaded when downloading packs

### ISSUE THREAD

**Issue by ffont** [MTG/freesound#914]
Currently when a pack is downloaded we simply add a row to the downloads table which indicates which pack has been downloaded. We should consider storing also information about which sounds were downloaded inside the pack. One option is to create a row in the downloads table for each sound (see discussion related in https://github.com/MTG/freesound/issues/840), but maybe is not the best one (it would require also changes in the attribution views, etc...). The other option is to add a new (JSON?) field to the download model and store list of sounds there. We can also add a new `PackDownload` model that stores this is a new table. Other options are welcome :)

**Comment by ffont** [comment]
After discussion in Hipchat and in person we agreed the following:

- We will create a `PackDownload` table in which we will store information about pack downloads including all the sounds downloaded for each pack.
- It is not yet clear whether we should store individual sounds as rows in the `PackDownload` (with the table having columns like `user_id,sound_id,license_id,pack_id,date`), or if we should store each pack download in a single row and then store the list of downloaded sounds and their licenses in a JSON field of that row (then columns would look like `user_id,pack_id,sounds_data,date`). As part of this issue, an experiment should be done comparing the two cases. Lets create both tables and evaluate their size and the efficiency of queries like "get all pack_ids downloaded by a user", "get sounds and licenses of a pack at the time user x downloaded it".
- Whatever strategy we choose, we'll have to write a command to copy pack downloads data from the current `Download` table to `PackDownload` table. This command might be slow and might take days to complete so we have to take that into account and think how are we going to avoid. If we don't do it right, it can happen that `PackDownload` ends up having rows whose date is not consecutive with the row id (like we are inserting information from new downloads as people download stuff and from old downloads as the command runs). We want to avoid for Postgres performance reasons. One possible solution is to have a flag in settings to disable storing new downloads to the `PackDownload` table, and have it set to false while we run the command to copy data. Then when the data is up to date we enable the flag.
- For this issue we are not making any changes to `Download` table. We will continue storing pack downloads into the current `Download` table, just as we are doing now, and not change the way we display lists of pack/sound downloads and attribution. This will happen in another issue after this one to "refactor downloads table".
- The last step of this issue should be to create a new one "Refactor Download table" which explains what to do next (rename the table, remove packs info from the table, changes in fronted, etc.)

**Comment by ffont** [comment]
Updating this issue after progress in #1043:

- We have already merged and deployed `PackDownload` and `PackDownloadSound` models in production, but these are currently empty and we are not yet using them to store pack downloads info. 
- We have a management command `copy_downloads` which fills the `PackDownload` table. 
- We don't have a command to fill `PackDownloadSound` because we don't really have that information from past pack downloads. Once at some point we implement a modal that after clicking on a pack download it shows the sounds that were downloaded as part of this pack and we don't have the info, we'll have to show "no info available for this pack".

What remains for this issue is:
1. Implement the logic to fill in  `PackDownload` and `PackDownloadSound` tables once pack downloads happen.
2. Use management command to fill `PackDownload` table.
3. Deploy what's implemented in point 1.

To do 2. and 3. we'll need some maintenance downtime while filling `PackDownload` so that no new downloads happen until we filled all the existing ones. We'll need to decide on the best strategy to minimize downtime. Firs of all we should know how long point 2. will take. That's something we could try in test.
